sm8250.c

```diff
@@ -0,0 +1,493 @@
+	{ "cam_cc_jpeg_clk", &gcc, 0x55, &cam_cc, 0x33 },
+	{ "cam_cc_mclk0_clk", &gcc, 0x55, &cam_cc, 0x1 },
+	{ "cam_cc_mclk1_clk", &gcc, 0x55, &cam_cc, 0x2 },
+	{ "cam_cc_mclk2_clk", &gcc, 0x55, &cam_cc, 0x3 },
+	{ "cam_cc_mclk3_clk", &gcc, 0x55, &cam_cc, 0x4 },
+	{ "cam_cc_mclk4_clk", &gcc, 0x55, &cam_cc, 0x5 },
+	{ "cam_cc_mclk5_clk", &gcc, 0x55, &cam_cc, 0x6 },
+	{ "cam_cc_mclk6_clk", &gcc, 0x55, &cam_cc, 0x7 },
+	{ "cam_cc_sbi_ahb_clk", &gcc, 0x55, &cam_cc, 0x4E },
+	{ "cam_cc_sbi_axi_clk", &gcc, 0x55, &cam_cc, 0x4D },
+	{ "cam_cc_sbi_clk", &gcc, 0x55, &cam_cc, 0x4A },
+	{ "cam_cc_sbi_cphy_rx_clk", &gcc, 0x55, &cam_cc, 0x4C },
+	{ "cam_cc_sbi_csid_clk", &gcc, 0x55, &cam_cc, 0x4B },
+	{ "cam_cc_sbi_ife_0_clk", &gcc, 0x55, &cam_cc, 0x4F },
+	{ "cam_cc_sbi_ife_1_clk", &gcc, 0x55, &cam_cc, 0x50 },
+	{ "cam_cc_sleep_clk", &gcc, 0x55, &cam_cc, 0x42 },
+
+	{ "disp_cc_mdss_ahb_clk", &gcc, 0x56, &disp_cc, 0x2B },
+	{ "disp_cc_mdss_byte0_clk", &gcc, 0x56, &disp_cc, 0x15 },
+	{ "disp_cc_mdss_byte0_intf_clk", &gcc, 0x56, &disp_cc, 0x16 },
+	{ "disp_cc_mdss_byte1_clk", &gcc, 0x56, &disp_cc, 0x17 },
+	{ "disp_cc_mdss_byte1_intf_clk", &gcc, 0x56, &disp_cc, 0x18 },
+	{ "disp_cc_mdss_dp_aux1_clk", &gcc, 0x56, &disp_cc, 0x25 },
+	{ "disp_cc_mdss_dp_aux_clk", &gcc, 0x56, &disp_cc, 0x20 },
+	{ "disp_cc_mdss_dp_link1_clk", &gcc, 0x56, &disp_cc, 0x22 },
+	{ "disp_cc_mdss_dp_link1_intf_clk", &gcc, 0x56, &disp_cc, 0x23 },
+	{ "disp_cc_mdss_dp_link_clk", &gcc, 0x56, &disp_cc, 0x1B },
+	{ "disp_cc_mdss_dp_link_intf_clk", &gcc, 0x56, &disp_cc, 0x1C },
+	{ "disp_cc_mdss_dp_pixel1_clk", &gcc, 0x56, &disp_cc, 0x1F },
+	{ "disp_cc_mdss_dp_pixel2_clk", &gcc, 0x56, &disp_cc, 0x21 },
+	{ "disp_cc_mdss_dp_pixel_clk", &gcc, 0x56, &disp_cc, 0x1E },
+	{ "disp_cc_mdss_edp_aux_clk", &gcc, 0x56, &disp_cc, 0x29 },
+	{ "disp_cc_mdss_edp_gtc_clk", &gcc, 0x56, &disp_cc, 0x2A },
+	{ "disp_cc_mdss_edp_link_clk", &gcc, 0x56, &disp_cc, 0x27 },
+	{ "disp_cc_mdss_edp_link_intf_clk", &gcc, 0x56, &disp_cc, 0x28 },
+	{ "disp_cc_mdss_edp_pixel_clk", &gcc, 0x56, &disp_cc, 0x26 },
+	{ "disp_cc_mdss_esc0_clk", &gcc, 0x56, &disp_cc, 0x19 },
+	{ "disp_cc_mdss_esc1_clk", &gcc, 0x56, &disp_cc, 0x1A },
+	{ "disp_cc_mdss_mdp_clk", &gcc, 0x56, &disp_cc, 0x11 },
+	{ "disp_cc_mdss_mdp_lut_clk", &gcc, 0x56, &disp_cc, 0x13 },
+	{ "disp_cc_mdss_non_gdsc_ahb_clk", &gcc, 0x56, &disp_cc, 0x2C },
+	{ "disp_cc_mdss_pclk0_clk", &gcc, 0x56, &disp_cc, 0xF },
+	{ "disp_cc_mdss_pclk1_clk", &gcc, 0x56, &disp_cc, 0x10 },
+	{ "disp_cc_mdss_rot_clk", &gcc, 0x56, &disp_cc, 0x12 },
+	{ "disp_cc_mdss_rscc_ahb_clk", &gcc, 0x56, &disp_cc, 0x2E },
+	{ "disp_cc_mdss_rscc_vsync_clk", &gcc, 0x56, &disp_cc, 0x2D },
+	{ "disp_cc_mdss_vsync_clk", &gcc, 0x56, &disp_cc, 0x14 },
+	{ "disp_cc_sleep_clk", &gcc, 0x56, &disp_cc, 0x37 },
+	{ "disp_cc_xo_clk", &gcc, 0x56, &disp_cc, 0x36 },
+
+	// { "apss_cc_debug_mux", &gcc, 0xE7 },
+	// { "cam_cc_debug_mux", &gcc, 0x55 },
+	// { "disp_cc_debug_mux", &gcc, 0x56 },
```

Comment:
Should probably leave these out.

Reply:
we leave these for informative purposes and comment them out so that they don't mess with the muxing process

Reply:
I did not do that for SM8150 as the downstream driver used a different mechanism to set the values, and did not include these "fake clock" entries. Should I add them? Looks like it's only from SM8250 onwards that they're there, but I'll definitely leave these in!

Reply:
well it's easier to compare if somebody didn't make a big oopsie if they're in, so it would be nice to add them.. though perhaps we could think about adding them in a somewhat "nicer" way than commented-out entries heh

Reply:
Yeah, in some way you want an array (per CC block!) like `[0xe7] = { "apss_cc_debug_mux" },` so that we can sort them on mux index _and_ have the compile scream on aforementioned oopsie duplicates.